### PR TITLE
emote_type refactor to bitflags

### LINKED ISSUE #65336: You can hear emotes while blind and deaf
## Reproduction:be blind and deaf and listen to a emote, 
unknown

### PR DESCRIPTION
**Pull request by Mooshimi**
## About The Pull Request

Fixes: #65336

## Why It's Good For The Game

<details><summary>You don't get to hear visible emotes if you are blind...</summary>

</details>

## Changelog

:cl:
fix: Blind people will now only be able to 'hear' audible emotes.
fix: Blind and deaf people will see and hear no emotes, for obvious reasons.
/:cl:

### CODE REVIEW

**Comment by tralezab** [comment]
perfect, now I can wave my hand in front of blind people's faces

**Comment by tralezab** [comment]
is death**gasp** not something you can hear? It should be

**Comment by tralezab** [comment]
lets make it audio in this pr 😤

**Comment by tralezab** [comment]
because otherwise it's kinda unacceptable

**Comment by tralezab** [comment]
could be both really, seizing up is visual, gasp is audio

**Comment by Mothblocks** [comment]
do we not have support for both visible and audible? i thought they were a bitfield

**Comment by Mooshimi** [comment]
we already have a var for if its audio or visible
we just didn't have one test to not show anything if they're blind and deaf

**Comment by Mooshimi** [comment]
couldn't i just make some emotes be this 
image

**Comment by Mothblocks** [comment]
That wouldn't work off the bat because it seems everything checks `==`.

You should change them to a bitfield so we can do `EMOTE_AUDIBLE | EMOTE_VISIBLE`

File: code/__DEFINES/flags.dm [hunk 1]
```diff
@@ -287,3 +287,7 @@ GLOBAL_LIST_INIT(bitflags, list(1, 2, 4, 8, 16, 32, 64, 128, 256, 512, 1024, 204
 #define FLORA_HERBAL (1 << 0)
 #define FLORA_WOODEN (1 << 1)
 #define FLORA_STONE (1 << 2)
+
+#define EMOTE_AUDIBLE (1<<0)
+#define EMOTE_VISIBLE (1<<1)
+#define EMOTE_IMPORTANT (1<<2)
```
Comment (Mothblocks) [inline code/__DEFINES/flags.dm:293]: Auto-doc all 3 of these

File: code/modules/mob/mob.dm [hunk 2]
```diff
@@ -207,6 +207,8 @@
 			else
 				msg = alt_msg
 				type = alt_type
+				if(type & MSG_AUDIBLE && !can_hear())
+					return
 
 		if(type & MSG_AUDIBLE && !can_hear())//Hearing related
```
Comment (Mothblocks) [inline code/modules/mob/mob.dm:213]: You early-return when this is the case earlier. Is this intentional?

Comment (Mooshimi) [reply code/modules/mob/mob.dm:213]: nope just realized it would skip over the second if all together, good catch

**Pull request merged**